pandas/core/arrays/datetimelike.py

```diff
@@ -123,6 +123,9 @@ def _check_compatible_with(
         Parameters
         ----------
         other
+        strict : bool, default False
+            For __setitem__ we may have stricter compatiblity resrictions than
```

Comment:
don’t we always need to check this for date times? why do u actually need a keyword

Reply:
ATM we dont use _check_compatible_with in the DTA comparison ops, but after this we will be able to.

Reply:
can you call this something else besides strict as this only applies to DTI, maybe check_tz (and ignore this elsewhere). i know strict is more generic, but also very confusing.

Reply:
changed "strict"->"setitem"